dlt/load/load.py

```diff
@@ -490,6 +490,28 @@ def run(self, pool: Optional[Executor]) -> TRunMetrics:
 
         return TRunMetrics(False, len(self.load_storage.list_normalized_packages()))
 
+    def maybe_trancate_staging_dataset(self, schema: Schema) -> None:
+        """
+        Truncate the staging dataset if one used,
+        and configuration requests truncation.
+
+        Args:
+            schema (Schema): Schema to use for the staging dataset.
+        """
+        try:
+            if self.config.truncate_staging_dataset:
+                with self.get_destination_client(schema) as client:
+                    if isinstance(client, WithStagingDataset):
+                        with client.with_staging_dataset():
+                            client.initialize_storage(truncate_tables=schema.data_table_names())
```

Comment:
here we need to pick only tables that are eligible for staging
- use `_extend_tables_with_table_chain` and pass schema.data_tables()
- select only tables that are eligible for staging (job_client.should_load_data_to_staging_dataset) - as table filter

and truncate returned tables

Reply:
Done